packages/ai-langchain/test/index.test.ts

```diff
@@ -0,0 +1,89 @@
+import { describe, it, expect, vi } from "vitest";
+import { FGARetriever } from "../src/retrievers/fga-retriever";
+import { OpenFgaClient, CredentialsMethod } from "@openfga/sdk";
+import { Document } from "@langchain/core/documents";
+import { BaseRetriever } from "@langchain/core/retrievers";
+
+describe("FGARetriever", () => {
+  process.env.FGA_CLIENT_ID = "client-id";
+  process.env.FGA_CLIENT_SECRET = "client-secret";
+
+  const mockRetriever = {
+    _getRelevantDocuments: vi.fn(),
+  } as unknown as BaseRetriever;
+
+  const mockBuildQuery = vi.fn((doc: Document, query: string) => ({
+    object: `doc:${doc.metadata.id}`,
+    relation: "viewer",
+    user: "user:user1",
+  }));
+
+  const mockClient = new OpenFgaClient({
```

Comment:
can we use some framework from mocking?

Reply:
we are using vitest for mocking the  `batchCheck` API calls itself, this client itself seems more straightforward like this since we want to test the creating of the retriever and OpenFGA SDK throws validation errors when these values are not passed.